### PR TITLE
Move mod/randprof to src/Module/RandomProfile

### PR DESCRIPTION
**Pull request by nupplaphil**
Part of #4090

### CODE REVIEW

File: src/Model/GContact.php [hunk 1]
```diff
@@ -1056,21 +1061,23 @@ public static function discoverGsUsers()
...
-			"SELECT `url` FROM `gcontact` WHERE `network` = '%s'
-					AND `last_contact` >= `last_failure`
-					AND `updated` > UTC_TIMESTAMP - INTERVAL 1 MONTH
-				ORDER BY rand() LIMIT 1",
-			DBA::escape(Protocol::DFRN)
-		);
+		$r = DBA::selectFirst('gcontact', ['url'], [
+			'`network` = ? 
+			AND `last_contact` >= `last_failure`  
+			AND	`updated` > ?',
```
Comment (MrPetovan) [inline src/Model/GContact.php]: Code standards: Misplaced tab.

File: src/Model/GContact.php [hunk 2]
```diff
@@ -1056,21 +1061,23 @@ public static function discoverGsUsers()
...
+			Protocol::DFRN,
+			DateTimeFormat::utc('now - 1 month'),
+		], ['order' => ['RAND()']]);
 
 		if (DBA::isResult($r)) {
-			return dirname($r[0]['url']);
```
Comment (MrPetovan) [inline src/Model/GContact.php]: Why did you remove the `dirname()` call?

File: src/Module/RandomProfile.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Friendica\Module;
+
+use Friendica\BaseModule;
+use Friendica\Model\Contact;
+use Friendica\Model\GContact;
+
+/**
+ * Redirects to a random profile of this node
```
Comment (MrPetovan) [inline src/Module/RandomProfile.php]: Records from the `gcontact` table doesn't contain only profiles from this node. A more accurate description would be `Redirects to a random Friendica profile this node knows about`

File: src/Model/GContact.php (hunk 2 above)
Comment (nupplaphil) [reply src/Model/GContact.php]: Because the `dirname` removed the last part of the url, so I got redirected to `friendica.local/profile` all the time. After removing it, I was correctly redirected to `friendica.local/profile/friendica1`
